src/directory/poll_watcher.rs

```diff
@@ -0,0 +1,73 @@
+use crate::core::META_FILEPATH;
+use crate::directory::error::OpenDirectoryError;
+use crate::directory::{WatchCallback, WatchCallbackList, WatchHandle};
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, UNIX_EPOCH};
+use std::fs;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+
+const POLL_INTERVAL_MS: u64 = 10;
+
+pub struct PollWatcher {
+    watcher_router: Arc<WatchCallbackList>,
+}
+
+#[derive(Hash)]
+struct State {
+    data: String,
+}
+
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+impl PollWatcher {
+    pub fn new(path: &Path) -> Result<Self, OpenDirectoryError> {
+        let watcher_router: Arc<WatchCallbackList> = Default::default();
+        let watcher_router_clone = watcher_router.clone();
+        let meta_path = path.to_owned().join(*META_FILEPATH);
+        thread::Builder::new()
+            .name("meta-file-watch-thread".to_string())
+            .spawn(move || {
+                let mut current_hash: u64 = Self::hash(&meta_path).unwrap_or(0);
+                let mut current_meta_time: u128 = Self::last_update(&meta_path).unwrap_or(0);
+                loop {
+                    let new_hash: u64 = Self::hash(&meta_path).unwrap_or(0);
```

Comment:
So if we receive an io::Error here, it will send the event once, and then never send an event again as long as an io::Error is returned when reading the file.

I think it is worth at least logging something here.

Reply:
If you can test what happens on a spurious `io::Error` using fail-rs it would be super cool.It is a bit of work, so if you can't just add a todo on this file :)